src/test/java/com/redhat/pantheon/servlet/ModuleJsonServletTest.java

```diff
@@ -0,0 +1,102 @@
+
+        // When
+        String query = servlet.getQuery(slingContext.request());
+
+        // Then
+        // make sure queries don't throw exceptions when executed against the JCR repository
+        assertDoesNotThrow(() -> slingContext.resourceResolver().queryResources(query, Query.JCR_SQL2));
+    }
+
+    @Test
+    void getQuery() {
+        // Given
+        ModuleJsonServlet servlet = new ModuleJsonServlet();
+        Map<String, Object> map = newHashMap();
+        map.put("locale", DEFAULT_MODULE_LOCALE.toString());
+        map.put("module_id", "jcr:uuid");
+        slingContext.request().setParameterMap(map);
+
+        // When
+        String query = servlet.getQuery(slingContext.request());
+
+        // Then
+        // make sure queries don't throw exceptions when executed against the JCR repository
+        assertDoesNotThrow(() -> slingContext.resourceResolver().queryResources(query, Query.JCR_SQL2));
+
+        // make sure query response is not null
+        assertNotNull(slingContext.response());
+    }
+
+
+    @Test
+    void resourceToMap() {
+        // Given
+        slingContext.create()
+                .resource("/content/repositories/repo/module/en_US/1",
+                        "jcr:primaryType", "pant:moduleRevision");
+        slingContext.create()
+                .resource("/content/repositories/repo/module/en_US/1/metadata",
+                        "jcr:title", "A title",
+                        "jcr:description", "A description");
+        slingContext.create()
+                .resource("/content/repositories/repo/module/en_US/1/content/asciidoc",
+                        "jcr:primaryType", "nt:file");
+        slingContext.create()
+                .resource("/content/repositories/repo/module/en_US/1/content/cachedHtml",
+                        "jcr:data", testHTML,
+                        "pant:hash", "2a0e2c43");
+        slingContext.resourceResolver().getResource("/content/repositories/repo/module/en_US").adaptTo(ModifiableValueMap.class)
+                .put("released", slingContext.resourceResolver().getResource("/content/repositories/repo/module/en_US/1").getValueMap()
+                        .get("jcr:uuid"));
+
+        registerMockAdapter(Module.class, slingContext);
+        ModuleJsonServlet servlet = new ModuleJsonServlet();
+        slingContext.request().setResource( slingContext.resourceResolver().getResource("/module") );
+
+        // When
+        Map<String, Object> map = servlet.resourceToMap(
+                slingContext.resourceResolver().getResource("/content/repositories/repo/module"));
+
+        // Then
+        assertTrue(map.containsKey("module"));
+        assertTrue(map.containsKey("status"));
+        assertTrue(map.containsKey("message"));
+        assertEquals((map.get("message")), "Module Found");
+        assertEquals((map.get("status")), "SUCCESS");
+        assertEquals((map.get("module").toString()), expectedModuleJson);
```

Comment:
These tests tend to be very brittle (i.e. the slightest format change will trigger a test failure, even if the json format itself is still valid and complete). I won't stop the PR for this one, but in the future lets change/stay away from these types of tests.

Reply:
I do agree with your point of not using this kind of hardcoded testing mocks.
However, IMO we should either move towards testing API endpoints against mock data or test if the responses contain required keys or not. This was an attempt to check against mock of real data to make sure the response is as expected.
We might need to take a call on which type tests we want to go with for API endpoints oing ahead.